go/pools/resource_pool.go

```diff
@@ -94,6 +98,25 @@ func NewResourcePool(factory Factory, capacity, maxCap int, idleTimeout time.Dur
 		rp.resources <- resourceWrapper{}
 	}
 
+	if prefillParallelism != 0 {
+		sem := sync2.NewSemaphore(prefillParallelism, 0 /* timeout */)
+		var wg sync.WaitGroup
+		for i := 0; i < capacity; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				_ = sem.Acquire()
+				defer sem.Release()
+				r, err := rp.Get(context.TODO())
+				if err != nil {
+					return
+				}
+				rp.Put(r)
+			}()
+		}
+		wg.Wait()
```

Comment:
Is it always necessary to wait for all the prefill actions to complete here?

I am somewhat reticent to add a ton of optionality here, but especially given the unbounded `context.TODO()`, this code as written might block startup for an arbitrary amount of time, no?

Reply:
This is the one part of this change that still worries me a bit -- as written this could potentially block startup forever, which seems like a bad idea. It would also be efficient to parallelize this with the other startup tasks.

At the same time if users really want to make sure the connection pool is prewarmed before serving queries... then that seems like the thing we should give them.

All in all though... I wonder if adding a bounded timeout of something like 30 seconds by default would be sufficient here?

Reply:
That seems better than waiting potentially forever.

Reply:
I've added a 30 second timeout. for prefilling the pool. We'll see how that works out.